hack/install.sh

```diff
@@ -134,6 +134,45 @@ temp_file="daytona-$RANDOM"
 # Ensure the temporary file is deleted on exit
 trap 'rm -f "$temp_file"' EXIT
 
+# Function to display a progress bar with a spinner
+progress_bar_with_spinner() {
+  local current_size=$1
+  local total_size=$2
+  local percent=$((current_size * 100 / total_size))
+  local bar_width=40
+  local filled=$((percent * bar_width / 100))
+  local empty=$((bar_width - filled))
+  local spinner="⠋⠙⠸⠴⠦⠧⠇⠏"
+  local spin_char=${spinner:$(($RANDOM % 8)):1}
+
+  printf "\r[%s] Downloading... [%-${bar_width}s] %d%%" "$spin_char" "$(printf "%0.s#" $(seq 1 $filled))$(printf "%0.s-" $(seq 1 $empty))" "$percent"
+}
+
+# Download the file while showing a progress bar with a spinner
+download_file() {
+  local url=$1
+  local output=$2
+
+  # Get the total size of the file to download
+  local total_size=$(curl -sI "$url" | grep -i Content-Length | awk '{print $2}' | tr -d '\r')
+
+  # Initialize a temporary file to hold the download
+  local temp_output=$(mktemp)
+
+  # Start downloading the file with `curl` and pipe the output to `dd` to track progress
+  curl -L -s "$url" -o "$temp_output" &
```

Comment:
![Screenshot 2024-08-28 at 15 13 59](https://github.com/user-attachments/assets/30834dcb-e47e-4575-a756-a6b0e141b9c2)

@varshith257 any reason why you decided to skip `-S` and `-f`? There are descriptions for those flags in the comment below.

P.S. That comment should also be moved up here.